src/deploy/osp_deployer/settings/gen_scripts/gen_prop_file.py

```diff
@@ -0,0 +1,260 @@
+                else:
+                    continue
+            else:
+                continue
+
+        if role_type in settings:
+            node_settings = settings[role_type]
+        else:
+            node_settings = {}
+
+        if node_id in node_settings:
+            node_id_settings = node_settings[node_id]
+        else:
+            node_id_settings = {}
+
+        if role_type.startswith("sah"):
+            node_id_settings['is_sah'] = 'true'
+        elif role_type.startswith("director"):
+            node_id_settings['is_director'] = 'true'
+        elif role_type.startswith("dashboard"):
+            node_id_settings['is_dashboard'] = 'true'
+        elif role_type.startswith("compute"):
+            node_id_settings['is_compute'] = 'true'
+        elif role_type.startswith("controller"):
+            node_id_settings['is_controller'] = 'true'
+        elif role_type.startswith("storage"):
+            node_id_settings['is_ceph_storage'] = 'true'
+        else:
+            continue
+
+        node_id_settings[node_setting_key] = value
+        node_settings[node_id] = node_id_settings
+        settings[role_type] = node_settings
+    return settings
+
+
+def generate_tmp_file(in_data, output_file, excluded_keys, use_service_tags):
+    fData = {}
+    i = 0
+    tmp_file = './tmpfile'
+    tmpFile = open(tmp_file, 'w')
+    tmpFile.write("[\n")
+
+    top_level_keys = in_data.keys()
+    key_count = len(top_level_keys)
+    for tlkey in top_level_keys:
+        nexlevel_keys = in_data.get(tlkey).keys()
+        i = i+1
+        for nexkey in nexlevel_keys:
+            odata = in_data.get(tlkey).get(nexkey)
+            # print "ODATA: {}".format(odata)
+
+            if use_service_tags and "idrac_ip" in odata:
+                del odata["idrac_ip"]
+            elif "service_tag" in odata:
+                del odata["service_tag"]
+
+            if "is_director" in odata:
+                del odata["storage_ip"]
+
+            if "is_controller" in odata:
+                del odata["storage_cluster_ip"]
+                del odata["provisioning_ip"]
+
+            if "is_compute" in odata:
+                del odata["storage_cluster_ip"]
+                del odata["provisioning_ip"]
+                del odata["public_api_ip"]
+
+            if "is_ceph_storage" in odata:
+                del odata["tenant_tunnel_ip"]
+                del odata["private_api_ip"]
+                del odata["provisioning_ip"]
+                del odata["public_api_ip"]
+
+            if excluded_keys is not None:
+                for exclude_key in excluded_keys:
+                    if exclude_key in odata:
+                        del odata[exclude_key]
+            fData[tlkey] = odata
+            output = json.dumps(fData.get(tlkey), indent=4, sort_keys=True)
+            tmpFile.write(output)
+            if i < key_count:
+                tmpFile.write(",\n")
+    tmpFile.write("\n]")
```

Comment:
Here, you can json.dumps with the list of odata and then write the string out to final file.